src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7440,8 +7440,15 @@ private SyntaxList<AttributeListSyntax> ParseStatementAttributeDeclarations()
             while (this.CurrentToken.Kind == SyntaxKind.OpenBracketToken)
                 ParseBracketedArgumentList();
 
-            // Check the next token to see if it indicates the `[...]` sequence we have is a term or not. 
-            var isCollectionExpression = this.CurrentToken.Kind is SyntaxKind.DotToken or SyntaxKind.QuestionToken or SyntaxKind.ExclamationToken;
+            // Check the next token to see if it indicates the `[...]` sequence we have is a term or not. This is the
+            // same set of tokens that ParsePostFixExpression looks for.
```

Comment:
are we going to remember to make a corresponding change here if another postfix operator is ever added to the language? :)

Reply:
I'll add comment